src/util/range.h

```diff
@@ -0,0 +1,348 @@
+/*******************************************************************\
+
+Module: Range
+
+Author: Romain Brenguier, romain.brenguier@diffblue.com
+
+\*******************************************************************/
+
+/// \file
+/// Ranges: pair of begin and end iterators.
+
+#ifndef CPROVER_UTIL_RANGE_H
+#define CPROVER_UTIL_RANGE_H
+
+#include <functional>
+#include <util/make_unique.h>
+
+/// Iterator for which dereference gives the result of some given
+/// function \c f applied on the underlying iterator.
+template <typename iteratort, typename outputt>
+class map_iteratort
+{
+public:
+  using difference_type = void;
+  using value_type = outputt;
+  using pointer = const outputt *;
+  using reference = const outputt &;
+  using iterator_category = std::forward_iterator_tag;
```

Comment:
These are public members that would deserve some documentation (in particular the use of `difference_type = void` makes me wonder).